service/header_service.go

```diff
@@ -291,6 +292,66 @@ func (hs *HeaderService) GetMerkleRootsConfirmations(
 	return hs.repo.Headers.GetMerkleRootsConfirmations(request, hs.merkleCfg.MaxBlockHeightExcess)
 }
 
+// LocateHeadersGetHeaders returns headers with given hashes.
+func (hs *HeaderService) LocateHeadersGetHeaders(locators []*chainhash.Hash, hashstop *chainhash.Hash) []*wire.BlockHeader {
```

Comment:
Overall very good job with this implementation.

The one thing that is missing and is rather hard challange is to add tests for this method.

Reply:
tests added